lib/plugins/rest.js

```diff
@@ -161,6 +161,40 @@ exports.register = function (server, options, next) {
 			});
 		}
 	});
+
+	server.route({
+		method  : "PUT",
+		path    : "/v1/instances/{instanceId}",
+		handler : function (request, reply) {
+			instances.getInstance({ id : request.params.instanceId })
+			.then(function (response) {
+
+				var instance = new Instance({
+					id    : request.params.instanceId,
+					ami   : request.payload.ami,
+					type  : request.payload.type,
+					state : request.payload.state,
+					uri   : request.payload.uri
+				});
+				var Constructor = instance.constructor;
+				var updatedInstance = new Constructor(
+					_.merge(
+						_.clone(instance),
+						{ revision : response.revision }
+					)
+				);
+				return instances.updateInstance(updatedInstance);
```

Comment:
This degrades the concurrency protection that `genesis` provides since it will make it look like all incoming requests are updating the current revision. Instead, the `revision` ID should be included on the incoming request payload. If the request payload includes the full `Instance` document, then we shouldn't need to call `getInstance()` either.